src/ansi2html.ts

```diff
@@ -0,0 +1,161 @@
+        case 7:
+          reverse = true;
+          break;
+        case 8: style.display = 'none'; break;
+        case 9: style['text-decoration'] = 'line-through'; break;
+        case 22:
+          delete style['font-weight'];
+          delete style['font-style'];
+          delete style['opacity'];
+          delete style['text-decoration'];
+          break;
+        case 23:
+          delete style['font-weight'];
+          delete style['font-style'];
+          delete style['opacity'];
+          break;
+        case 24:
+          delete style['text-decoration'];
+          break;
+        case 27:
+          reverse = false;
+          break;
+        case 30:
+        case 31:
+        case 32:
+        case 33:
+        case 34:
+        case 35:
+        case 36:
+        case 37:
+          fg = ansiColors[code - 30];
+          break;
+        case 39:
+          fg = defaultColors?.fg;
+          break;
+        case 40:
+        case 41:
+        case 42:
+        case 43:
+        case 44:
+        case 45:
+        case 46:
+        case 47:
+          bg = ansiColors[code - 40];
+          break;
+        case 49:
+          bg = defaultColors?.bg;
+          break;
+        case 53: style['text-decoration'] = 'overline'; break;
+        case 90:
+        case 91:
+        case 92:
+        case 93:
+        case 94:
+        case 95:
+        case 96:
+        case 97:
+          fg = brightAnsiColors[code - 90];
+          break;
+        case 100:
+        case 101:
+        case 102:
+        case 103:
+        case 104:
+        case 105:
+        case 106:
+        case 107:
+          bg = brightAnsiColors[code - 100];
+          break;
+      }
+    } else if (text) {
+      let token = escapeHTML(text);
+      const isBold = style['font-weight'] === 'bold';
+      if (isBold)
+        token = `<b>${token}</b>`;
+      const isItalic = style['font-style'] === 'italic';
+      if (isItalic)
+        token = `<i>${token}</i>`;
+      const hasOpacity = style['opacity'] === '0.8';
+      if (hasOpacity)
+        token = `<span style='color:#666;'>${token}</span>`;
+      const color = reverse ? (bg || '#000') : fg;
+      if (color)
+        token = `<span style='color:${color};'>${token}</span>`;
+      const backgroundColor = reverse ? fg : bg;
+      if (backgroundColor)
+        token = `<span style='background-color:${backgroundColor};'>${token}</span>`;
+      tokens.push(token);
+    }
+  }
+  return tokens.join('');
+}
+
+const ansiColors: Record<number, string> = {
+  0: '#000',
+  1: '#f14c4c',
+  2: '#73c991',
+  3: '#ffcc66',
+  4: '#44a8f2',
+  5: '#b084eb',
+  6: '#afdab6',
+  7: '#fff',
+};
+
+const brightAnsiColors: Record<number, string> = {
+  0: '#808080',
+  1: '#f14c4c',
+  2: '#73c991',
+  3: '#ffcc66',
+  4: '#44a8f2',
+  5: '#b084eb',
+  6: '#afdab6',
+  7: '#fff',
+};
+
+function escapeHTML(text: string): string {
+  return text.replace(/[&"<>\s\n]/g, c => ({
+    ' ': '&nbsp;',
```

Comment:
this won't match other whitespaces, such as \t. Also, do we want to map all whitespaces to non-breaking?

Reply:
This has not changed

Reply:
hmm, this is what I see upstream:

```js
function escapeHTML(text: string): string {
  return text.replace(/[&"<>]/g, c => ({ '&': '&amp;', '"': '&quot;', '<': '&lt;', '>': '&gt;' }[c]!));
}
```